Migrate xpallets-assets-registrar to pallet v2

VegeBun-csj opened the pull request.

VegeBun-csj committed:
migrate xpallets-assets-registrar pallet v2

VegeBun-csj committed:
fix some error

0xrjman commented on xpallets/assets-registrar/src/lib.rs:
```diff
@@ -19,118 +19,59 @@ pub mod weights;
-        /// Asset id list for each Chain.
-        pub AssetIdsOf get(fn asset_ids_of): map hasher(twox_64_concat) Chain => Vec<AssetId>;
+    #[pallet::config]
+    /// The module's config trait.
```
```suggestion
    /// The pallet's config trait.
```

0xrjman commented on xpallets/assets-registrar/src/lib.rs:
```diff
@@ -219,13 +162,104 @@ decl_module! {
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    /// Event for the XAssetRegistrar Pallet
```
```suggestion
    /// Event for the XAssetRegistrar Pallet
    #[pallet::event]
    #[pallet::generate_deposit(pub(super) fn deposit_event)]
```

0xrjman reviewed (commented):
I suggest rearranging the order of pallet components as implied in the upgrade guide:
`#[pallet::config]`
`#[pallet::pallet]`
`#[pallet::call]`
...

VegeBun-csj committed:
adjust the order of pallet component

VegeBun-csj committed:
adjust the documents position

liuchengxu commented on xpallets/assets-registrar/src/lib.rs:
```diff
@@ -300,17 +334,30 @@ impl<T: Config> Module<T> {
+    ) -> Result<(), String> {
+        <Self as GenesisBuild<T>>::assimilate_storage(self, storage)
+    }
+}
```
This is not neccessary, the compile issue should be resolved in another way.

```suggestion
```

alannotnerd replied:
impl `GenesisBuild` in std

alannotnerd committed:
Impl `GenesisBuild` while in std

alannotnerd committed:
Merge remote-tracking branch 'origin/develop' into assets-registrar

alannotnerd committed:
Update mock due to no generic in GenesisConfig

liuchengxu committed:
Clean up

- Remove unused import.
- Replace Module with Pallet.
- Restore the log line in asssets-registrar that was accidently deleted.

liuchengxu committed:
Fix format
